src/libostree/ostree-repo-checkout.c

```diff
@@ -126,19 +122,11 @@ write_regular_file_content (OstreeRepo            *self,
 
   if (mode != OSTREE_REPO_CHECKOUT_MODE_USER)
     {
-      int res;
-      do
-        res = fchown (fd,
-                      g_file_info_get_attribute_uint32 (file_info, "unix::uid"),
-                      g_file_info_get_attribute_uint32 (file_info, "unix::gid"));
-      while (G_UNLIKELY (res == -1 && errno == EINTR));
-      if (G_UNLIKELY (res == -1))
+      if (TEMP_FAILURE_RETRY (fchown (fd, g_file_info_get_attribute_uint32 (file_info, "unix::uid"),
+                                      g_file_info_get_attribute_uint32 (file_info, "unix::gid"))) < 0)
         return glnx_throw_errno (error);
```

Comment:
This is more of a general comment rather than on this specific code, though after just reviewing the libglnx PR https://github.com/GNOME/libglnx/pull/43 (where more call sites were converted to throw errors prefixed with the syscall), it does seem like it would be nice to make it the standard coding style, I think? Might help debugging and diagnosing user-reported issues.